src/sundials/sundials_memory.c

```diff
@@ -203,23 +203,44 @@ int SUNMemoryHelper_CopyAsync(SUNMemoryHelper helper, SUNMemory dst,
 
 int SUNMemoryHelper_Destroy(SUNMemoryHelper helper)
 {
-  if (helper->ops->destroy == NULL)
+  if (!helper) return 0;
+
+  if (helper->ops)
   {
-    if (helper->content != NULL)
+    if (helper->ops->destroy(helper))
```

Comment:
Shouldn't this be `if (helper->ops->destroy)` so that you can check the nullity of the pointer before calling the function?